src/components/Classes/components/JoinClassModal.js

```diff
@@ -1,89 +1,107 @@
-import React from "react";
-import * as fetch from "../../../lib/fetch.js";
-import { Redirect } from "react-router-dom";
-
-import { Button, Container, Row, Col, FormGroup, Label, Input } from "reactstrap";
-
-import ReactModal from "react-modal";
-
-class JoinClassModal extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      wid: "",
-      disableSubmit: false,
-      error: "",
-      redirect: false,
-    };
-  }
-
-  closeModal = () => {
-    if (this.props.onClose && {}.toString.call(this.props.onClose) === "[object Function]") {
-      this.props.onClose();
+import React, { useState } from 'react';
+// import * as fetch from "../../../lib/fetch.js";
+import ReactModal from 'react-modal';
+import { Redirect } from 'react-router-dom';
+
+import {
+  Button, Container, Row, Col, FormGroup, Label, Input,
+} from 'reactstrap';
+
+const JoinClassModal = (props) => {
+  const [wid, setWid] = useState('');
+  // const [_disableSubmit, setDisableSubmit] = useState("")
```

Comment:
hm we should be able to leave this in without errors because it starts with `_`

See https://eslint.org/docs/rules/no-unused-vars#argsignorepattern + https://github.com/uclaacm/TeachLAFrontend/blob/master/.eslintrc.js#L46

Reply:
Also, note that the default value should be a bool, just for semantic reasons.

Reply:
it still shows unused variable when I uncomment it, seems like the documentation is saying that you can ignore only if the variable is an argument, but this is a useState variable.

Reply:
I think you can just check for the flag on line 58, which should resolve the issue.

Reply:
ok nice catch, you can just add `varsIgnorePattern: '^_' ` to the `.eslintrc.js` if you'd like! lmk if you need help w this :))

Also to @lumisphere902, not really sure what 
> I think you can just check for the flag on line 58, which should resolve the issue. 
means?

Reply:
@reginawang3495 yah can you show me where to put varsIgnorePattern: '^_' ? 
I'm kinda confused where to put it

Reply:
I was just thinking that if we actually used the flag like this: https://github.com/uclaacm/TeachLAFrontend/pull/716/files#r748608755, we could avoid the eslint issue. The eslintrc change would go here: https://github.com/uclaacm/TeachLAFrontend/blob/master/.eslintrc.js#L46

Reply:
I think it’d be good to put in the eslint anyhow!

Reply:
Yeah I added disableSubmit at the line and the eslint didn't check for it, so i think it should be fine